frontend/src/pages/BatchSend.vue

```diff
@@ -0,0 +1,729 @@
+    </q-form>
+  </q-page>
+</template>
+
+<script lang="ts">
+// --- External imports ---
+import { computed, defineComponent, onMounted, ref, watch } from 'vue';
+import { QForm, QInput, QSelect } from 'quasar';
+import { RandomNumber, utils as umbraUtils } from '@umbracash/umbra-js';
+// --- Components ---
+import BaseInput from 'components/BaseInput.vue';
+import BaseSelect from 'components/BaseSelect.vue';
+import BaseTooltip from 'components/BaseTooltip.vue';
+import ConnectWallet from 'components/ConnectWallet.vue';
+// --- Store ---
+import useSettingsStore from 'src/store/settings';
+import useWalletStore from 'src/store/wallet';
+// --- Other ---
+import { tc } from 'src/boot/i18n';
+import { txNotify } from 'src/utils/alerts';
+import {
+  BigNumber,
+  BigNumberish,
+  Contract,
+  formatUnits,
+  getAddress,
+  hexValue,
+  MaxUint256,
+  parseUnits,
+  // TransactionResponse,
+  Zero,
+} from 'src/utils/ethers';
+import { humanizeTokenAmount, humanizeMinSendAmount, humanizeArithmeticResult } from 'src/utils/utils';
+import { generatePaymentLink, parsePaymentLink } from 'src/utils/payment-links';
+import { Provider, TokenInfoExtended, supportedChains } from 'components/models';
+import { ERC20_ABI } from 'src/utils/constants';
+import { toAddress } from 'src/utils/address';
+
+interface SendData {
+  receiver: string | undefined;
+  token: TokenInfoExtended | null;
+  amount: string;
+  recipientIdBaseInputRef: any;
+}
+
+// Umbra.js is expected types
+interface newSendBatch {
+  token: string;
+  amount: BigNumberish;
+  address: string;
+}
+
+function useBatchSendForm() {
+  const { advancedMode } = useSettingsStore();
+  const {
+    balances,
+    chainId,
+    currentChain,
+    // getTokenBalances,
+    isLoading,
+    NATIVE_TOKEN,
+    provider,
+    setNetwork,
+    signer,
+    tokens: tokenList,
+    umbra,
+    userAddress,
+  } = useWalletStore();
+
+  // Helpers
+  const sendFormRef = ref<QForm>();
+  const isSending = ref(false);
+  const advancedAcknowledged = ref(false);
+  const showAdvancedSendWarning = ref(false);
+  const acknowledgeSendRisk = ref(false);
+
+  // Form refs for triggering validation via form items' `validate()` method.
+  const recipientIdBaseInputRef = ref<InstanceType<typeof BaseInput> | null>(null);
+  const tokenBaseSelectRef = ref<InstanceType<typeof BaseSelect> | null>(null);
+  const humanAmountBaseInputRef = ref<InstanceType<typeof BaseInput> | null>(null);
+
+  // Form parameters.
+  const Sends = ref<SendData[]>([{ receiver: '', token: null, amount: '', recipientIdBaseInputRef: null }]);
+  const recipientId = ref<string>();
+  const useNormalPubKey = ref(false);
+  const token = ref<TokenInfoExtended | null>();
+  const humanAmount = ref<string>();
+  const isValidForm = ref(false);
+  const isValidRecipientId = ref(true); // for showing/hiding bottom space (error message div) under input field
+  const toll = ref<BigNumber>(Zero);
```

Comment:
Yes, most of the code is taken from the `Send` page. Sorry this was meant to be just a quick review of the UI shown in the video, but it was very helpful to have you look at the code early. 

I've made the PR target branch the `Batch Send Umbra-js` branch so it has cleared out the umbra-js related commits. I haven't made any changes to compared with the existing PR.

I'll work making this page as simple as it can be and see where we could maybe refactor/modularize, and if we should split it up.